packages/datadog-instrumentations/src/tedious.js

```diff
@@ -28,7 +28,9 @@ addHook({ name: 'tedious', versions: ['>=1.0.0'] }, tedious => {
     const connectionConfig = this.config
 
     return asyncResource.runInAsyncScope(() => {
-      startCh.publish({ queryOrProcedure, connectionConfig })
+      const payload = { queryOrProcedure, connectionConfig }
+      startCh.publish(payload)
+      queryParent[queryField] = payload.sql
```

Comment:
maybe we could pass `queryParent` and `queryField` on the payload to the start event so we can do this on the subscriber side when doing `injectDbmQuery`, instead of the instrumentation having to know about it and set it here (i think it should be fine, since they're all references, but correct me if i'm wrong). i don't mind much either way. wdyt?

Reply:
You might be onto something. Ideally `getQueryOrProcedure` could do the mutation but the publish expects the original object.